src/App.tsx

```diff
@@ -15,36 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  None = '',
+}
+
 export const App: React.FC = () => {
+  const [items, setItems] = useState<string[]>(goodsFromServer);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [initialItems] = useState<string[]>(goodsFromServer);
+
+  function handleAlphabetSort() {
+    if (sortType === SortType.Alphabet && !isReversed) {
+      return;
+    }
+
+    const sortedItems = [...items].sort((a, b) => a.localeCompare(b));
+
+    if (isReversed) {
+      sortedItems.reverse();
+    }
+
+    setItems(sortedItems);
+    setSortType(SortType.Alphabet);
+  }
+
+  function handleLengthSort() {
+    if (sortType === SortType.Length && !isReversed) {
+      return;
+    }
+
+    const sortedItems = [...items].sort((good1, good2) => {
+      if (good1.length === good2.length) {
+        return good1.localeCompare(good2);
+      }
+
+      return good1.length - good2.length;
+    });
+
+    if (isReversed) {
+      sortedItems.reverse();
+    }
```

Comment:
you can create only one function that will accept our array, sort value and isReversed

and this function will return the sorted array that wi can reuse everywhere
```
const visibleGoods = getVisibleGoods(goodsFromServer, sortType, isReversed)
```